typing/typecore.ml

```diff
@@ -4967,8 +5018,15 @@ and type_cases
     | _ -> true
   in
   let outer_level = get_current_level () in
-  with_local_level_iter_if may_contain_gadts begin fun () ->
+  with_local_level_iter_if create_inner_level begin fun () ->
   let lev = get_current_level () in
+  let allow_modules =
+    if may_contain_modules
+    then
+      let scope = create_scope () in
```

Comment:
Actually this `create_scope` is not really useful: you could as well use `lev`, since `create_scope` returns `current_level`.
Calling `create_scope` has the extra side-effect of raising the `current_level` for subsequent uses, but I don't think it would make a difference here: those two levels are not distinguished inside this `with_local_level_iter_if`.

Reply:
Also you wrote in your blob that the responsability for checking escape is with the one who defines `allow_modules`. It would be worth indicating here that this is going to be checked together with GADT-induced existentials, hence no need for extra code.

Reply:
I took the proposed suggestion and added a comment, thanks.